qiskit/circuit/quantumcircuit.py

```diff
@@ -6785,32 +6788,47 @@ def box(
                 qc.box(body_1, [4, 5, 6, 7, 8], [])
 
         Args:
-            body: if given, the :class:`QuantumCircuit` to use as the box's body in the explicit
-                construction.  Not given in the context-manager form.
+            body_or_annotations: the first positional argument is unnamed.  If a
+                :class:`QuantumCircuit` is passed positionally, it is immediately used as the body
+                of the box, and ``qubits`` and ``clbits`` must also be specified.  If not given, or
+                if given an iterable of :class:`.Annotation` objects, the context-manager form of
+                this method is triggered.
             qubits: the qubits to apply the :class:`.BoxOp` to, in the explicit form.
             clbits: the qubits to apply the :class:`.BoxOp` to, in the explicit form.
             label: an optional string label for the instruction.
             duration: an optional explicit duration for the :class:`.BoxOp`.  Scheduling passes are
                 constrained to schedule the contained scope to match a given duration, including
                 delay insertion if required.
             unit: the unit of the ``duration``.
+            annotations: any :class:`.Annotation` objects the box should have.  When this method is
+                used in context-manager form, this argument can instead be passed as the only
+                positional argument.
         """
-        if isinstance(body, QuantumCircuit):
+        if isinstance(body_or_annotations, QuantumCircuit):
             # Explicit-body form.
+            body = body_or_annotations
             if qubits is None or clbits is None:
                 raise CircuitError("When using 'box' with a body, you must pass qubits and clbits.")
+            if annotations is Ellipsis:
+                annotations = []
             return self.append(
-                BoxOp(body, duration=duration, unit=unit, label=label),
+                BoxOp(body, duration=duration, unit=unit, label=label, annotations=annotations),
                 qubits,
                 clbits,
                 copy=False,
             )
+        if body_or_annotations is ...:
+            annotations = () if annotations is ... else annotations
+        elif annotations is not ...:
+            raise TypeError("QuantumCircuit.box() got multiple values for argument 'annotations'")
```

Comment:
This might be slightly unclear. Could we mention that annotations were passed both positionally as the first argument and via the 'annotations' kwarg?